### PR TITLE
[AppConfig] Javadoc and readme revision

### PR DESCRIPTION
**Pull request by mssfang**
- updated pricing javadoc to a link instead of fixed working
- fixes: #10320 readme

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 1]
```diff
@@ -612,8 +612,10 @@
 
     /**
      * Lists chronological/historical representation of {@link ConfigurationSetting} resource(s). Revisions are provided
-     * in descending order from their {@link ConfigurationSetting#getLastModified() lastModified} date. Revisions expire
-     * after a period of time. The service maintains change history for up to 7 days.
+     * in descending order from their {@link ConfigurationSetting#getLastModified() lastModified} date.
+     * Revisions expire after a period of time.
+     * <p>See <a href="https://azure.microsoft.com/en-us/pricing/details/app-configuration/">Pricing</a>
+     * for more information.</p>
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java:618]: Why does this link to pricing? Is there a better page of documentation which discusses how long a revision will be maintained?

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java [hunk 2]
```diff
@@ -398,8 +398,11 @@ public ConfigurationSetting setReadOnly(String key, String label, boolean isRead
 
     /**
      * Lists chronological/historical representation of {@link ConfigurationSetting} resource(s). Revisions are provided
-     * in descending order from their {@link ConfigurationSetting#getLastModified() lastModified} date. Revisions expire
-     * after a period of time. The service maintains change history for up to 7 days.
+     * in descending order from their {@link ConfigurationSetting#getLastModified() lastModified} date.
+     * Revisions expire after a period of time.
+     * <p>See <a href="https://azure.microsoft.com/en-us/pricing/details/app-configuration/">Pricing</a>
+     * for more information.</p>
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java:404]: I think we should leave the last sentence as a single sentence, no need to break it up due to the hyperlink.

```suggestion
     * Revisions expire after a period of time, see <a href="https://azure.microsoft.com/en-us/pricing/details/app-configuration/">this documentation</a>
     * for more information.</p>
```

File: sdk/appconfiguration/azure-data-appconfiguration/README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ Use the client library for App Configuration to create and manage application co
 
 ### Prerequisites
 
-- Java Development Kit (JDK) with version 8 or above
+- [Java Development Kit (JDK)][azure_jdk] with version 8 or above
```
Comment (srnagar) [inline sdk/appconfiguration/azure-data-appconfiguration/README.md:15]: We should be compatible with any distribution of JDK 8+ and not just `azure_jdk`. Is there a reason for linking to Azure JDK distribution here?

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java (hunk 1 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java:618]: key History in pricing. Price = free, 7 days; Price = standard, 30 days. Keep it as link as we don't want to set it to fixed.

File: sdk/appconfiguration/azure-data-appconfiguration/README.md (hunk 3 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/README.md:15]: https://github.com/Azure/azure-sdk-for-java/issues/10320

First section suggestted.

**Pull request merged**